nuxeo-drive-client/tests/test_synchronization_suspend.py

```diff
@@ -32,6 +27,46 @@ def test_basic_synchronization_suspend(self):
         self.assertEqual(len(remote.get_children_info(self.workspace_1)), 4)
         self.assertTrue(self.engine_1.get_queue_manager().is_paused())
 
+    def test_synchronization_local_watcher_paused_when_offline(self):
+        """ NXDRIVE-680: fix unwanted local upload when offline. """
+
+        local = self.local_client_1
+        remote = self.remote_document_client_1
+        engine = self.engine_1
+
+        # Create one file locally and wait for sync
+        engine.start()
+        self.wait_sync(wait_for_async=True)
+        local.make_file('/', 'file1.txt', content=b'42')
+        self.wait_sync()
+
+        # Checks
+        self.assertTrue(remote.exists('/file1.txt'))
+        self.assertTrue(local.exists('/file1.txt'))
+
+        # Simulate offline mode (no more network for instance)
+        engine.get_queue_manager().suspend()
+
+        # Create a bunch of files locally
+        local.make_folder('/', 'files')
+        for num in range(20):
```

Comment:
Shouldn't we check with more than 50 files, value of the `_windows_queue_threshold` that triggers the full scan?